apollo-normalized-cache/src/main/java/com/apollographql/apollo/cache/normalized/ApolloStoreOperation.kt

```diff
@@ -0,0 +1,102 @@
+package com.apollographql.apollo.cache.normalized
+
+import com.apollographql.apollo.exception.ApolloException
+import java.util.concurrent.Executor
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * Apollo store operation to be performed.
+ *
+ *
+ * This class is a wrapper around operation to be performed on [ApolloStore]. Due to the fact that any operation
+ * can potentially include SQLite instruction, any operation on [ApolloStore] must be performed in background
+ * thread. Use [.enqueue] to schedule such operation in the dispatcher with a callback to get results.
+ *
+ *
+ * @param <T> result type for this operation
+</T> */
+abstract class ApolloStoreOperation<T> protected constructor(private val dispatcher: Executor?) {
+  private val callback = AtomicReference<Callback<T>?>()
+  private val executed = AtomicBoolean()
+  protected abstract fun perform(): T
```

Comment:
Let's put 1 space between properties and the function
```suggestion

  protected abstract fun perform(): T
```